presto-thrift-testing-udf-server/src/main/java/com/facebook/presto/udf/thrift/EchoFirstInputThriftUdfService.java

```diff
@@ -50,7 +51,11 @@ public EchoFirstInputThriftUdfService(BlockEncodingSerde blockEncodingSerde)
         ThriftUdfPage result;
         switch (inputs.getPageFormat()) {
             case PRESTO_THRIFT:
-                result = thriftPage(ImmutableList.of(inputs.getThriftBlocks().get(0)));
+                PrestoThriftPage thriftPage = inputs.getThriftPage();
+                if (thriftPage.getThriftBlocks().isEmpty()) {
+                    throw new UnsupportedOperationException("No input to echo");
```

Comment:
Since this is an example to help people write their own UDF service, maybe we should show a working case for no-arg UDFs, since there might be some confusion on how to properly handle it? It's not obvious what needs to be done for that situation. For example, need to write back results equal to number of input rows, etc.